Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -373,169 +258,156 @@ class HistoryPanel: SiteTableViewController, HomePanel {
+        // First section is for Sync/recently closed and has no title.
+        guard section > 0 else {
+            return nil
+        }
+
+        // Ensure there are rows in this section.
+        guard groupedSites.numberOfItemsForSection(section - 1) > 0 else {
+            return nil
+        }
+
+        switch section - 1 {
+        case 0:
+            return Strings.TableDateSectionTitleToday
+        case 1:
+            return Strings.TableDateSectionTitleYesterday
+        case 2:
+            return Strings.TableDateSectionTitleLastWeek
+        case 3:
+            return Strings.TableDateSectionTitleLastMonth
         default:
-            assertionFailure("Invalid history section \(section)")
+            assertionFailure("Invalid History section \(section)")
+        }
+
+        return nil
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAt: indexPath)
+        cell.accessoryType = .none
+
+        // First section is reserved for Sync/recently closed.
+        guard indexPath.section > 0 else {
+            cell.imageView?.layer.borderWidth = 0
+            return indexPath.row == 0 ? configureRecentlyClosed(cell, for: indexPath) : configureSyncedTabs(cell, for: indexPath)
+        }
+
+        return configureSite(cell, for: indexPath)
+    }
+
+    // MARK: - UITableViewDelegate
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        // First section is reserved for Sync/recently closed.
+        guard indexPath.section > 0 else {
+            tableView.deselectRow(at: indexPath, animated: true)
+            return indexPath.row == 0 ? navigateToRecentlyClosed() : navigateToSyncedTabs()
+        }
+
+        if let site = siteForIndexPath(indexPath), let url = URL(string: site.url) {
+            if let homePanelDelegate = homePanelDelegate {
+                homePanelDelegate.homePanel(self, didSelectURL: url, visitType: VisitType.typed)
+            }
+            return
+        }
+        print("Error: No site or no URL when selecting row.")
+    }
+
+    override func tableView(_ tableView: UITableView, willDisplayHeaderView view: UIView, forSection section: Int) {
+        if let header = view as? UITableViewHeaderFooterView {
+            header.textLabel?.textColor = UIColor.theme.tableView.headerTextDark
+            header.contentView.backgroundColor = UIColor.theme.tableView.headerBackground
         }
-        return title
     }
 
     override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        if section == 0 {
+        // First section is for Sync/recently closed and its header has no view.
+        guard section > 0 else {
+            return nil
+        }
+
+        // Ensure there are rows in this section.
+        guard groupedSites.numberOfItemsForSection(section - 1) > 0 else {
```

Comment:
maybe this whole function will just be`if Section(section).siteCountFor(groupedSites: groupedSites) < 1 return nil`, or is that too odd?

``` enum Section ...
func siteCountFor(groupedSites: groupedSites): Int {
  // if self is not a site section return zero
}